File: solutions/dart/collatz-conjecture/1/lib/collatz_conjecture.dart
class CollatzConjecture {
  int steps(int number) {
    if (number < 1) {
      throw ArgumentError('Only positive integers are allowed');
    }
    int count = 0;
    while (number != 1) {
      if (number % 2 == 0) {
        number = number ~/ 2;
      } else {
        number = 3 * number + 1;
      }
      count++;
    }
    return count;
  }
}

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    step=0
    if number<=0:
        raise ValueError("Only positive integers are allowed")

    if number==1: 
        return step
    
    while number!=1:
        step+=1
        if number%2==0:
            number=number/2
        else:
            number =3*number+1

    return step
